packages/frontend/src/store/timeEntries.ts

```diff
@@ -219,3 +239,105 @@ function updateTimeEntries(state: State, paramEntries: FrontendTimentrie[]) {
   }
   state.timeEntries = newTimeEntries;
 }
+
+function getRelevantMonths(): Date[] {
+  const today = new Date();
+  const lastMonth = new Date(today.getFullYear(), today.getMonth() - 1);
+  const twoMonthsAgo = new Date(
+    lastMonth.getFullYear(),
+    lastMonth.getMonth() - 1
+  );
+  const relevantMonths = [twoMonthsAgo, lastMonth, today];
+  return relevantMonths;
+}
+
+function filterTimeEntriesByRelevantMonths(
+  allTimeEntriesAllTasks: TimeEntriesDateFormated[][]
+) {
+  return allTimeEntriesAllTasks.map(timeEntriesSingleTask => {
+    return timeEntriesSingleTask.filter(x => {
+      return getRelevantMonths().some(y => {
+        if (y.getMonth() == x.date.getMonth()) {
+          return y;
+        }
+      });
+    });
+  });
+}
+
+function convertStringToDateAndValueToNumber(
+  timeEntries: FrontendTimentrie[] | null
+) {
+  var newArray: TimeEntriesDateFormated[] = timeEntries
+    ? timeEntries.map((x: FrontendTimentrie) => {
+        let newObj = {
+          id: x.id,
+          value: Number(x.value),
+          taskId: x.taskId,
+          date: new Date(x.date),
+        };
+        return newObj;
+      })
+    : [];
+  return newArray;
+}
+
+function splitIntoTasks(
+  timeEntriesFormated: TimeEntriesDateFormated[]
+): TimeEntriesDateFormated[][] {
+  var uniqueTasks = [...new Set(timeEntriesFormated.map(x => x.taskId))];
+
+  var hoursPrTask = uniqueTasks.map(x => {
+    return timeEntriesFormated.filter(y => y.taskId === x);
+  });
+
+  return hoursPrTask;
+}
+
+function monthSum(x: any): any {
+  let initArray = getRelevantMonths().map(x => {
+    return {
+      date: x,
+      value: 0,
+    };
+  });
+  var value = x.reduce((acc: any, cur: any) => {
+    let existItem = acc.find(
+      (x: TimeEntriesDateFormated) => cur.date.getMonth() == x.date.getMonth()
+    );
+    if (existItem) {
+      existItem.value += cur.value;
+    } else {
+      acc.push(cur);
+    }
+    return acc;
+  }, initArray);
+  return value;
+}
```

Comment:
1. Gjøre om `let` og `var` til `const`.
2. Skrive om `reduce` til en for of loop. reduce er etter min mening den minst lesbare funksjonen som finnes siden man må lese slutten av funksjonen for å få vite hva den starter med.
3. Gi `x` et ordentlig navn og en interface. `x` parameteret her kan alt for lett bli forvekslet med en av de 2 andre parameterene som også har blitt kalt `x`.
4. Gjerne skrive en test.